src/devices/wifi/nm-iwd-manager.c

```diff
@@ -104,7 +104,10 @@ psk_agent_dbus_method_cb (GDBusConnection *connection,
 	if (!nm_streq0 (g_dbus_object_manager_client_get_name_owner (omc), sender))
 		goto return_error;
 
-	g_variant_get (parameters, "(&o)", &network_path);
+	if (!strcmp (method_name, "RequestUserPassword"))
+		g_variant_get (parameters, "(&os)", &network_path, &username);
```

Comment:
use `&s` to not leak `username`. As you don't use the parameter, I think you could just pass `NULL` instead.

Reply:
Good point, fixed.